core/src/main/scala/com/salesforce/op/dsl/RichTextFeature.scala

```diff
@@ -747,22 +747,27 @@ trait RichTextFeature {
      * Converts a sequence of [[ID]] features into a vector keeping the top K occurrences of each feature,
      * along with an extra column per feature indicating how many values were not in the top K.
      *
-     * @param others     Other [[ID]] features to include in pivot
+     * @param maxCardinality     Maximum Cardinality for a categorical
      * @param topK       How many values to keep in the vector
      * @param minSupport Min times a value must occur to be retained in pivot
      * @param cleanText  If true, ignores capitalization and punctuations when grouping categories
      * @param trackNulls keep an extra column that indicated if feature was null
+     * @param others     Other [[ID]] features to include
      * @return The vectorized features
      */
     def vectorize
     (
+      maxCardinality: Int,
```

Comment:
since this is a mandatory argument - this is a breaking change. do we want it to be a mandatory one or can we have a default value? @mweilsalesforce @Jauntbox

Reply:
I guess a Default Value might be ok

Reply:
I'd imagine we'd want to set the max cardinality pretty high (~1000?), since even when we pivot we're still using a topK. We should really make this (and SmartTextVectorizer) have thresholds based on data size, but can do that in a separate PR. 

For example, we could do what percentage of the values are covered by the topK values rather than a strict cardinality limit. We could also do something like (unique count)/(row count) > threshold, instead of unique count > threshold.

Reply:
Yes, this was discussed with @kinfaikan long time ago.